main/command/src/main/scala/sbt/MainLoop.scala

```diff
@@ -65,6 +65,10 @@ object MainLoop {
       val newLogging = state.globalLogging.newLogger(out, logBacking)
       transferLevels(state, newLogging)
       val loggedState = state.copy(globalLogging = newLogging)
+      def isInteractive = System.console() != null
+      def hasShell = state.remainingCommands contains "shell"
+      if (isInteractive && !hasShell)
+        state.log info "!!! Executing in batch mode !!! For better performance, hit [ENTER] to remain in the sbt shell"
```

Comment:
I found that without my "!!!" even **I** who implemented the feature couldn't see it.

That's why I agreed with Lars that the issue was in the choice of using info. Move it to warn allowed me to drop the !!! workaround, make the message shorter, and also (as you can see from the screenshots below) benefit from `[warn]` being in yellow.

We have only 1 axis (error/warn/info/debug) on which to attribute semantics. I don't know you, but if I IRL see someone run `sbt compile` I would want to warn them.

> Yes, your build is going to compile slower, but it's not like it's destroying your code.

It's not an error, just a warning.